src/commands/postDeploy/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,176 @@
+            }
+
+            if (Date.now() > detectorTimeoutMs) {
+                if (showOutput) {
+                    const noIssuesFound: string = `Diagnosing for "${node.root.client.siteName}" has timed out.`;
+                    ext.outputChannel.appendLine(noIssuesFound);
+                    context.telemetry.properties.failureCount = '0';
+                }
+                return undefined;
+            }
+
+            // update the new end time to be now
+            // tslint:disable-next-line: no-unsafe-any
+            requestOptions.qs.endTime = new Date().toISOString();
+            detectorErrorMessage = await getLinuxDetectorError(context, requestOptions, deployResultTime, node.root.client.fullName);
+
+            // poll every minute
+            await delay(1000 * 60);
+        } while (!detectorErrorMessage);
+
+        if (showOutput) {
+            // this detector has more valuable information to show in the portal than 'LinuxContainerStartFailure'
+            const linuxAppDownId: string = 'LinuxAppDown';
+            await ext.ui.showWarningMessage(detectorErrorMessage, { title: 'Open in Portal' });
+            await openInPortal(node.root, `${node.root.client.id}/troubleshoot`, { queryPrefix: `websitesextension_ext=asd.featurePath%3Ddetectors%2F${linuxAppDownId}` });
+            context.telemetry.properties.didClick = 'true';
+        }
+    });
+}
+
+async function getLinuxDetectorError(context: IActionContext, requestOptions: WebResource, deployResultTime: Date, siteName: string): Promise<string | undefined> {
+    const detectorResponse: string = await requestUtils.sendRequest<string>(requestOptions);
+    const responseJson: detectorResponseJSON = <detectorResponseJSON>JSON.parse(detectorResponse);
+
+    const datasets: detectorDataset[] = responseJson.properties.dataset;
+    const vsCodeIntegration: string = 'Latest time seen by detector. To be used in VSCode integration.';
+    const timestampTable: detectorTable | undefined = findTableByRowValue(datasets, vsCodeIntegration);
+
+    // if we can't find the timestamp, exit and try again
+    if (!timestampTable) {
+        return undefined;
+    }
+
+    const timestamp = new Date(getValuesByColumnName(timestampTable, ColumnName.value)[0]);
+    if (validateTimestamp(timestamp, deployResultTime)) {
+
+        const criticalErrorTable: detectorTable | undefined = findTableByRowValue(datasets, 'Critical');
+        const failureCountTable: detectorTable | undefined = findTableByColumnName(datasets, ColumnName.failureCount);
+
+        // if the criticalError table or failureCounts aren't defined, we can't display anything to the user
+        if (!criticalErrorTable || !failureCountTable) {
+            return undefined;
+        }
+
+        const errorMessages: string[] = getValuesByColumnName(criticalErrorTable, ColumnName.value);
+        const failureCount: string = getValuesByColumnName(failureCountTable, ColumnName.failureCount)[0];
```

Comment:
So how often do we expect multiple rows vs. just the first row? Since we only ever show the first one... Is it possible the first row will have 0 failures and not be useful? Can we somehow find the most useful row?

Should probably add telemetry for how many rows there are.

Also maybe just change `getValuesByColumnName` to always return the first row. I feel like that would be simpler than always doing `[0]`